### DISCUSSION TITLE
Launching an additional program.

### DISCUSSION THREAD

**Discussion by henassy84** [kimmknight/remoteapptool#116] [Q&A]
Can you tell me how to run multiple programs in one session?
Currently, when opening an rdp session, one program opens. For example, there are two independent databases, both of them are opened by one program, how can I open two shortcuts of the same program in an rdp session at the RemoteAppTool level, but with connection to different databases?

**Comment by MrBrianGale** [comment]
As far as I know, this isn't possible (with some exceptions, but those are application-based exceptions) based on how remote apps work. One benefit of remote apps is that they are session based. What I mean is that if you start the remote app up on your laptop then the battery dies, you can go over to your desktop and start the same app up in the exact same session state it was in before. As long as you don't "exit" the app or log off the server, the app will remain in the last state it was in on that session. There are exceptions (such as if auto-logout is enabled), but what you are requesting is not a thing that remote app tool can do; it would need to be implemented in the application layer.

Remote Apps don't allow you to run the same app twice in most circumstances. The exception is with apps that allow starting multiple copies of itself like excel where you could open multiple excel files and it'll happily let you have multiple windows.
Basically, you would need the app to handle this, not a remote app tool. Remote App Tool is mostly just a front-end for the Remote App protocol portion of RDP. It's smarter than just a front end as it makes setting up Remote Apps a LOT easier than Microsoft's tool AND it works on systems that Microsoft doesn't support Remote Apps (Windows 10 for example), but any limitations of the RDP protocol are limitations of this app too.

Now, a workaround to get what you want - have 2 copies of the application on the server and create 2 different RDP files pointing to the 2 different locations on disk of the server. Not ideal, but it gets around the limitation of RDP.

**Reply by henassy84** [reply]
Thank you very much!
